File: api/src/main/java/org/algo/invest/core/quotes/IndicesQuotes.java
package org.algo.invest.core.quotes;

import org.algo.invest.model.Industry;
import org.algo.invest.model.QuoteSymbolMetaData;

import java.util.LinkedHashMap;
import java.util.Map;

public class IndicesQuotes {

    public static Map<String, QuoteSymbolMetaData> getQuotes() {

        Map<String, QuoteSymbolMetaData> symbolNameMapping = new LinkedHashMap<String, QuoteSymbolMetaData>();

        symbolNameMapping.put("EURUSD=X", new QuoteSymbolMetaData("EURUSD=X","EUR/USD", "", Industry.NONE));
        symbolNameMapping.put("USDCNY=X", new QuoteSymbolMetaData("USDCNY=X","USD/CNY", "", Industry.NONE));
        symbolNameMapping.put("BTC-USD", new QuoteSymbolMetaData("BTC-USD","BTC/USD", "", Industry.NONE));
        symbolNameMapping.put("GC=F", new QuoteSymbolMetaData("GC=F","GOLD/USD", "", Industry.NONE));
        symbolNameMapping.put("CL=F", new QuoteSymbolMetaData("CL=F","Crude Oil/USD", "", Industry.NONE));
        symbolNameMapping.put("^VIX", new QuoteSymbolMetaData("^VIX","Volatility Index", "", Industry.NONE));
        symbolNameMapping.put("^GDAXI", new QuoteSymbolMetaData("^GDAXI","DAX", "", Industry.NONE));
        symbolNameMapping.put("^MDAXI", new QuoteSymbolMetaData("^MDAXI","MDAX", "", Industry.NONE));
        symbolNameMapping.put("^SDAXI", new QuoteSymbolMetaData("^SDAXI","SDAX", "", Industry.NONE));
        symbolNameMapping.put("^TECDAX", new QuoteSymbolMetaData("^TECDAX","TecDAX", "", Industry.NONE));
        symbolNameMapping.put("^DJI", new QuoteSymbolMetaData("^DJI","Dow Jones", "", Industry.NONE));
        symbolNameMapping.put("^IXIC", new QuoteSymbolMetaData("^IXIC","Nasdaq Comp", "", Industry.NONE));
        symbolNameMapping.put("^NDX", new QuoteSymbolMetaData("^NDX","Nasdaq 100", "", Industry.NONE));
        symbolNameMapping.put("^NDXT", new QuoteSymbolMetaData("^NDXT","Nasdaq 100 Tech", "", Industry.NONE));
        symbolNameMapping.put("^GSPC", new QuoteSymbolMetaData("^GSCP","S&P 500", "", Industry.NONE));
        symbolNameMapping.put("NQ=F", new QuoteSymbolMetaData("NQ=F","Nasdaq 100 Future", "", Industry.NONE));
        symbolNameMapping.put("YM=F", new QuoteSymbolMetaData("YM=F","Dow Jones Future", "", Industry.NONE));
        symbolNameMapping.put("ES=F", new QuoteSymbolMetaData("ES=F","S&P 500 Future", "", Industry.NONE));
        symbolNameMapping.put("^N225", new QuoteSymbolMetaData("^N225","Nikkei", "", Industry.NONE));
        symbolNameMapping.put("000001.SS", new QuoteSymbolMetaData("000001.SS","Shanghai Comp", "", Industry.NONE));
        symbolNameMapping.put("^HSI", new QuoteSymbolMetaData("^HSI","Hang Seng HK", "", Industry.NONE));
        symbolNameMapping.put("^HSCE", new QuoteSymbolMetaData("^HSCE","Hang Seng China", "", Industry.NONE));

        return symbolNameMapping;
    }
}

File: api/src/main/java/org/algo/invest/core/quotes/IndicesQuotes.kt
package org.algo.invest.core.quotes

import org.algo.invest.model.Category
import org.algo.invest.model.QuoteSymbolMetaData
import java.util.LinkedHashMap
import org.algo.invest.model.Industry

object IndicesQuotes {
    val quotes: Map<String, QuoteSymbolMetaData>
        get() {
            val symbolNameMapping: MutableMap<String, QuoteSymbolMetaData> = LinkedHashMap()

            symbolNameMapping["EURUSD=X"] = QuoteSymbolMetaData("EURUSD=X", "EUR/USD", "", Industry.INDICES)
            symbolNameMapping["USDCNY=X"] = QuoteSymbolMetaData("USDCNY=X", "USD/CNY", "", Industry.INDICES)
            symbolNameMapping["^TNX"] = QuoteSymbolMetaData("^TNX", "Treasury 10y", "", Industry.INDICES)
            symbolNameMapping["BTC-USD"] = QuoteSymbolMetaData("BTC-USD", "BTC/USD", "", Industry.INDICES)
            symbolNameMapping["GC=F"] = QuoteSymbolMetaData("GC=F", "GOLD/USD", "", Industry.INDICES)
            symbolNameMapping["CL=F"] = QuoteSymbolMetaData("CL=F", "Crude Oil/USD", "", Industry.INDICES)
            symbolNameMapping["^VIX"] = QuoteSymbolMetaData("^VIX", "Volatility Index", "", Industry.INDICES)
            symbolNameMapping["^GDAXI"] = QuoteSymbolMetaData("^GDAXI", "DAX", "", Industry.INDICES)
            symbolNameMapping["^MDAXI"] = QuoteSymbolMetaData("^MDAXI", "MDAX", "", Industry.INDICES)
            symbolNameMapping["^SDAXI"] = QuoteSymbolMetaData("^SDAXI", "SDAX", "", Industry.INDICES)
            symbolNameMapping["^TECDAX"] = QuoteSymbolMetaData("^TECDAX", "TecDAX", "", Industry.INDICES)
            symbolNameMapping["^DJI"] = QuoteSymbolMetaData("^DJI", "Dow Jones", "", Industry.INDICES)
            symbolNameMapping["^IXIC"] = QuoteSymbolMetaData("^IXIC", "Nasdaq Comp", "", Industry.INDICES)
            symbolNameMapping["^NDX"] = QuoteSymbolMetaData("^NDX", "Nasdaq 100", "", Industry.INDICES)
            symbolNameMapping["^NDXT"] = QuoteSymbolMetaData("^NDXT", "Nasdaq 100 Tech", "", Industry.INDICES)
            symbolNameMapping["^GSPC"] = QuoteSymbolMetaData("^GSCP", "S&P 500", "", Industry.INDICES)
            symbolNameMapping["NQ=F"] = QuoteSymbolMetaData("NQ=F", "Nasdaq 100 Future", "", Industry.INDICES)
            symbolNameMapping["YM=F"] = QuoteSymbolMetaData("YM=F", "Dow Jones Future", "", Industry.INDICES)
            symbolNameMapping["ES=F"] = QuoteSymbolMetaData("ES=F", "S&P 500 Future", "", Industry.INDICES)
            symbolNameMapping["^N225"] = QuoteSymbolMetaData("^N225", "Nikkei", "", Industry.INDICES)
            symbolNameMapping["000001.SS"] = QuoteSymbolMetaData("000001.SS", "Shanghai Comp", "", Industry.INDICES)
            symbolNameMapping["^HSI"] = QuoteSymbolMetaData("^HSI", "Hang Seng HK", "", Industry.INDICES)
            symbolNameMapping["^HSCE"] = QuoteSymbolMetaData("^HSCE", "Hang Seng China", "", Industry.INDICES)

            symbolNameMapping["EUNL.DE"] = QuoteSymbolMetaData("EUNL.DE", "iShares Core MSCI World", "A0RPWH", Industry.INDICES, Category.ETF)
            symbolNameMapping["L0CK.DE"] = QuoteSymbolMetaData("L0CK.DE", "iShares Digital Security", "A2JMGE", Industry.INDICES, Category.ETF)
            symbolNameMapping["IQQ6.DE"] = QuoteSymbolMetaData("IQQ6.DE", "iShares Developed Markets Property Yield", "A0LEW8", Industry.INDICES, Category.ETF)
            symbolNameMapping["2B78.DE"] = QuoteSymbolMetaData("2B78.DE", "iShares Healthcare Innovation", "A2ANH2", Industry.INDICES, Category.ETF)
            symbolNameMapping["IBB"] = QuoteSymbolMetaData("IBB", "iShares Nasdaq Biotechnology", "A2DWAW", Industry.INDICES, Category.ETF)
            symbolNameMapping["ICLN"] = QuoteSymbolMetaData("ICLN", "iShares Global Clean Energy", "", Industry.INDICES, Category.ETF)
            symbolNameMapping["ESPO"] = QuoteSymbolMetaData("ESPO", "VanEck Vectors Video Gaming and eSports", "", Industry.INDICES, Category.ETF)

            return symbolNameMapping
        }
}